sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreCheckpointFormatV2Suite.scala

```diff
@@ -211,6 +218,317 @@ class RocksDBStateStoreCheckpointFormatV2Suite extends StreamTest
     }
   }
 
+  val changelogEnabled =
+    "spark.sql.streaming.stateStore.rocksdb.changelogCheckpointing.enabled" -> "true"
+  val changelogDisabled =
+    "spark.sql.streaming.stateStore.rocksdb.changelogCheckpointing.enabled" -> "false"
+  val ckptv2 = SQLConf.STATE_STORE_CHECKPOINT_FORMAT_VERSION.key -> "2"
+
+  case class checkpointV2CompatibilityTest(
+    configsForFirstRun: Seq[(String, String)],
+    configsForSecondRun: Seq[(String, String)]
+  ) {
+    override def toString: String = "first run - " +
+      s"changeLogEnabled: ${configsForFirstRun(0)._2}, ckpt ver: ${configsForFirstRun(1)._2}" +
+      s" - second run - changeLogEnabled: ${configsForSecondRun(0)._2}, " +
+      s"ckpt ver: ${configsForSecondRun(1)._2}"
+  }
+
+  val testConfigSetups = Seq(
+    // Enable and disable changelog under ckpt v2
+    checkpointV2CompatibilityTest(Seq(changelogEnabled, ckptv2), Seq(changelogEnabled, ckptv2)),
+    checkpointV2CompatibilityTest(Seq(changelogDisabled, ckptv2), Seq(changelogDisabled, ckptv2))
```

Comment:
Here is an example how likely the code can look like this (not necessarily the best way but just to give some ideas). We can have something like
```
case class CheckpointConfig(ifEnableChangelog: bool, checkpointVersion: int)
```
and we build a helper class like so that we can run like this:
```
test(...) {
  ...
  testWithCheckpointConfig(config1) {
     // run queries ....
  }
  testWithCheckpointConfig(config2) {
     // run queries ....
  }
}
```